core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -400,30 +427,27 @@ public void persist(final AlluxioURI path, final ScheduleAsyncPersistencePOption
 
   @Override
   public FileInStream openFile(AlluxioURI path)
-      throws FileDoesNotExistException, IOException, AlluxioException {
+      throws FileDoesNotExistException, OpenDirectoryException, FileIncompleteException,
+      IOException, AlluxioException {
     return openFile(path, OpenFilePOptions.getDefaultInstance());
   }
 
   @Override
   public FileInStream openFile(AlluxioURI path, OpenFilePOptions options)
-      throws FileDoesNotExistException, IOException, AlluxioException {
+      throws FileDoesNotExistException, OpenDirectoryException, FileIncompleteException,
+      IOException, AlluxioException {
     checkUri(path);
-    return rpc(client -> {
-      AlluxioConfiguration conf = mFsContext.getPathConf(path);
-      URIStatus status = client.getStatus(path,
-          FileSystemOptions.getStatusDefaults(conf).toBuilder()
-              .setAccessMode(Bits.READ)
-              .setUpdateTimestamps(options.getUpdateLastAccessTime())
-              .build());
-      if (status.isFolder()) {
-        throw new FileDoesNotExistException(
-            ExceptionMessage.CANNOT_READ_DIRECTORY.getMessage(status.getName()));
-      }
-      OpenFilePOptions mergedOptions = FileSystemOptions.openFileDefaults(conf)
-          .toBuilder().mergeFrom(options).build();
-      InStreamOptions inStreamOptions = new InStreamOptions(status, mergedOptions, conf);
-      return new FileInStream(status, inStreamOptions, mFsContext);
-    });
+    URIStatus status = getStatus(path, options);
+    return openFileInStream(path, status, options);
+  }
+
+  protected FileInStream openFileInStream(AlluxioURI path, URIStatus status,
```

Comment:
I don't think we should break up the method in this class in order to better modularize for a class that inherits from this one.

Reply:
this is just a utility function, without it, any extension to `BaseFileSystem` that wants to override `openFile` need to duplicate this logic.